docs/choreolib/auto-factory.md

```diff
@@ -1,109 +1,268 @@
-                              routine.kill();
-                              return new Pose2d();
-                            }))
-                .andThen(trajectory.cmd())
-                .withName("twoPieceAuto entry point"));
-
-    trajectory.atTime("intake").onTrue(intake.extend());
-    trajectory.atTime("shoot").onTrue(shooter.launch());
+// Creates a new routine with the name "exampleRoutine"
+AutoRoutine routine = autoFactory.newRoutine("exampleRoutine");
+```
+
+The "entrance" to all routines is the `AutoRoutine.running()` trigger. You should bind the first command you want to execute to this trigger.
+
+```java
+routine.running().onTrue(Commands.print("Started the routine!"));
+```
+
+Traditional triggers can also be used in conjunction with an auto routine. However, they will experience "hygiene" issues if not used correctly, causing them to be active even when the routine is not. You must either:
+
+- Use `AutoRoutine.observe()` to ensure the trigger is only active when the routine is running
+- Conjoin a trigger to one created by `AutoRoutine` or `AutoTrajectory`
+
+```java
+Trigger myTrigger = new Trigger(() -> condition);
+
+// Safe
+routine.observe(myTrigger).onTrue(Commands.print("Foo"));
+routine.running().and(myTrigger).onTrue(Commands.print("Bar"));
+
+// Unsafe
+myTrigger.onTrue(Commands.print("Foo"));
+myTrigger.and(routine.running()).onTrue(Commands.print("Bar"));
+```
+
+Trajectories can be loaded using `AutoRoutine.trajectory()`, which will return an `AutoTrajectory` ([Java](/api/choreolib/java/choreo/auto/AutoTrajectory.html)). The `AutoTrajectory` class exposes multiple triggers for you to attach reactive logic to, as well as `AutoTrajectory.cmd()` for scheduling the trajectory.
+
+```java
+public AutoRoutine pickupAndScoreAuto() {
+    AutoRoutine routine = autoFactory.newRoutine("pickupAndScore");
+
+    // Load the routine's trajectories
+    AutoTrajectory pickupTraj = routine.trajectory("pickupGamepiece");
+    AutoTrajectory scoreTraj = routine.trajectory("scoreGamepiece");
+
+    // When the routine begins, reset odometry and start the first trajectory (1)
+    routine.running().onTrue(pickupTraj.resetOdometry().andThen(pickupTraj.cmd()));
+
+    // Starting at the event marker named "intake", run the intake
+    pickupTraj.atPose("intake").onTrue(intakeSubsystem.intake());
+
+    // When the trajectory is done, start the next trajectory
+    pickupTraj.done().onTrue(scoreTraj.cmd());
+
+    // While the trajectory is active, prepare the scoring subsystem
+    scoreTraj.active().whileTrue(scoringSubsystem.getReady());
+
+    // When the trajectory is done, score
+    scoreTraj.done().onTrue(scoringSubsystem.score());
 
     return routine;
-  }
+}
+```
+
+1. You should always reset your robot's odometry to the start of the first trajectory being followed in an autonomous routine. `AutoTrajectory.resetOdometry()` will accomplish this, setting the robot's pose to the start of the trajectory.
```

Comment:
Is this true? some teams might have vision good enough they don't need to
I do like this as general tip though

Reply:
Teams should only be seeding their field position with vision if they really know what they're doing, and as far as I know the number of teams that actually did this in 2024 could probably be counted on my fingers. We also don't know what AprilTag layouts will look like in future seasons, and how optimized they will be for pose estimation from wherever robots start on the field. I could modify the wording to mention vision but idk how much value it adds, if you think it's important I can make the change